test/async_cache_test.dart

```diff
@@ -136,4 +132,25 @@ void main() {
     await cache.invalidate();
     expect(await cache.fetchStream(call).toList(), ['Called 3']);
   });
+
+  test('should cancel a cached stream without effecting others', () async {
+    Stream<String> call() {
+      return new Stream<String>.fromIterable(['1', '2', '3']);
+    }
+    expect(cache.fetchStream(call).toList(), completion(['1', '2', '3']));
+    expect(await cache.fetchStream(call).first, '1');
+  });
+
+  test('should pause a cached stream without effecting others', () async {
+    Stream<String> call() {
+      return new Stream<String>.fromIterable(['1', '2', '3']);
+    }
+    StreamSubscription sub;
+    sub = cache.fetchStream(call).listen((event) {
+      if (event == '1') {
+        sub.pause();
+      }
+    });
```

Comment:
Wrap this callback in `expectAsync` so we can be sure the stream is actually getting paused.